src/featureManager.ts

```diff
@@ -30,15 +35,68 @@ export class FeatureManager {
 
     // If multiple feature flags are found, the first one takes precedence.
     async isEnabled(featureName: string, context?: unknown): Promise<boolean> {
-        const featureFlag = await this.#provider.getFeatureFlag(featureName);
-        if (featureFlag === undefined) {
-            // If the feature is not found, then it is disabled.
-            return false;
+        const result = await this.#evaluateFeature(featureName, context);
+        return result.enabled;
+    }
+
+    async getVariant(featureName: string, context?: ITargetingContext): Promise<Variant | undefined> {
+        const result = await this.#evaluateFeature(featureName, context);
+        return result.variant;
+    }
+
+    async #assignVariant(featureFlag: FeatureFlag, context: ITargetingContext): Promise<{
+        variant: VariantDefinition | undefined;
+        reason: VariantAssignmentReason;
+    }> {
+        // user allocation
+        if (featureFlag.allocation?.user !== undefined) {

```

Comment:
I don't like how nested this section of code gets. Could we do terminators instead? For example this check would become:

```typescript
if (featureFlag.allocation?.user === undefined) {
    return;
}
...
```

Reply:
Not sure if I get the point, but if featureFlag.allocation.user is undefined, we continue to check featureFlag.allocation.group and try group allocation, instead of a direct return.

Reply:
It's just personal preference that I like to avoid too many tabs/nested checks. You're right and we couldn't do it on the first check since the code needs to continue. We could do it on the `isTargetedUser` check could be terminated to remove at least one level.

```typescript
        if (featureFlag.allocation?.user !== undefined) {
            for (const userAllocation of featureFlag.allocation.user) {
                if (!isTargetedUser(context.userId, userAllocation.users)) {
                    continue;
                }
                const variant = featureFlag.variants?.find(v => v.name == userAllocation.variant);
                if (variant !== undefined) {
                    return { variant, reason: VariantAssignmentReason.User }; 
                } else {
                    console.warn(`Variant ${userAllocation.variant} not found for feature ${featureFlag.id}.`);
                }
            }
        }
```

Reply:
helper function extracted in 63cdd43 , now it looks like below, and I made a slight change to avoid using continue:
```ts
       if (featureFlag.allocation?.user !== undefined) {
            for (const userAllocation of featureFlag.allocation.user) {
                if (isTargetedUser(context.userId, userAllocation.users)) {
                    return getVariantAssignment(...);
                }
            }
        }
```